### PR TITLE
[usage] Implement CollectUsage

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Moves the Usage Controller into an RPC called `CollectUsage`. This change gives us the following:
* Better decoupling between usage and billing
* Better observability through existing gRPC method monitoring
* More flexible structure to move services around
* Simplifies scheduled job to compute usage and update billing

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/10785

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview with-billing

### CODE REVIEW

File: components/usage/pkg/apiv1/usage.go [hunk 1]
```diff
@@ -88,6 +97,212 @@ func (us *UsageService) ListBilledUsage(ctx context.Context, in *v1.ListBilledUs
...
+		nowFunc: time.Now,
+	}
+}
+
+type ReportGenerator struct {
+	conn    *gorm.DB
+	pricer  *controller.WorkspacePricer
+	nowFunc func() time.Time
+}
+
```
Comment (andrew-farries) [inline components/usage/pkg/apiv1/usage.go:196]: Should we remove this `ReportGenerator` to it's own file now.? This `usage.go` is now rather large.

Comment (easyCZ) [reply components/usage/pkg/apiv1/usage.go:196]: Sure, that's easy enough :)

**Review by andrew-farries — APPROVED** [review]
/hold for splitting out `UsageReport`

File: components/usage/pkg/apiv1/usage.go (hunk 1 above)
Comment (easyCZ) [reply components/usage/pkg/apiv1/usage.go]: Done and rebased. Also moved the pricer into `apiv1` as it didn't make sense in the controller anymore

**Pull request merged**